### PR TITLE
LibUPnP 1.16 compat

### PR DESCRIPTION
**Pull request by whyman**

### CODE REVIEW

File: src/server.cc [hunk 1]
```diff
@@ -160,19 +162,21 @@ void Server::run()
 
     log_debug("Registering with UPnP...");
     ret = UpnpRegisterRootDevice2(
+         UPNP_LIB_ARG
```
Comment (neheb) [inline src/server.cc:165]: tab?

Comment (whyman) [reply src/server.cc:165]: Probably, dont worry, its WIP still!

**Comment by whyman** [comment]
Conan why you no work

**Comment by neheb** [comment]
Need rebase.

File: src/server.cc [hunk 2]
```diff
@@ -160,39 +161,46 @@ void Server::run()
 
     log_debug("Registering with UPnP...");
     ret = UpnpRegisterRootDevice2(
-        UPNPREG_BUF_DESC,
+        UPNP_LIB_ARG
+            UPNPREG_BUF_DESC,
         deviceDescription.c_str(),
-        deviceDescription.length() + 1,
+        size_t(deviceDescription.length()) + 1,
```
Comment (neheb) [inline src/server.cc:167]: length() is already size_t.

**Comment by whyman** [comment]
> Need rebase.

Needs to actually work :laughing: :cry:

**Comment by neheb** [comment]
That I don't know about.

BTW am I wrong in seeing that they're migrating to C++?

**Comment by whyman** [comment]
Looks like the 1.16 branch is discontinued

**Pull request closed without merging**